test/functional/page_objects/visualize_page.js

```diff
@@ -611,12 +611,12 @@ export function VisualizePageProvider({ getService, getPageObjects, updateBaseli
       await PageObjects.header.waitUntilLoadingHasFinished();
     }
 
-    async toggleOtherBucket() {
-      return await find.clickByCssSelector('vis-editor-agg-params:not(.ng-hide) input[name="showOther"]');
+    async toggleOtherBucket(agg = 2) {
+      return await testSubjects.click(`aggregationEditor${agg} otherBucketSwitch`);
     }
 
-    async toggleMissingBucket() {
-      return await find.clickByCssSelector('vis-editor-agg-params:not(.ng-hide) input[name="showMissing"]');
+    async toggleMissingBucket(agg = 2) {
+      return await testSubjects.click(`aggregationEditor${agg} missingBucketSwitch`);
```

Comment:
I'm confused why you need to update selector here- would it be enough to use `testSubjects.click('otherBucketSwitch')`? I looked at the scenarios you're testing and they seem to only have one Terms aggregation each.

Reply:
We have test scenarios where there is a sub-bucket:
![image](https://user-images.githubusercontent.com/31325372/56952407-fa212300-6b42-11e9-842f-8574ec7908e8.png)
This aggregation will have an id `3` and we also have aggregation with id `2` which also have same aggs (but it is in hidden state, that is why previously it was possible with `vis-editor-agg-params:not(.ng-hide) input[name="showMissing"]`). We could still rely on `:not(.ng-hide)` attribute, but we are working on removing angularjs code, so it will be invalid in the near future.

It is not the only place where we use an approach with aggregation id :
![slack](https://user-images.githubusercontent.com/31325372/56953347-0a3a0200-6b45-11e9-9401-31c22e26131d.png)